### DISCUSSION TITLE
Delete based on a join?

### DISCUSSION THREAD

**Discussion by Foxsterc** [simolus3/drift#2785] [Q&A] [answered]
Hi,

Is is possible to delete based on joining X tables with criteria from both?

Thanks

**Comment by simolus3** [comment, marked as answer]
It's not possible to have joins on delete statements, but in most cases it's possible to come up with a subquery to use in `WHERE` that does the same thing. E.g. if you wanted to do `DELETE FROM t1 JOIN t2 ON t2.foo = t1.bar WHERE t2.another = 3;`, you could instead do

```dart
final deleteStmt = delete(t1);
deleteStmt.where((t1) {
  final subquery = db.select(t2)
    ..where((t2) => t2.foo.equalsExp(t1.bar) & t2.another.equals(3));
  return existsQuery(subquery);
});

await deleteStmt.go();
```

**Comment by Foxsterc** [comment]
That's perfect! I hadn't thought of using a subquery in drift.

Thanks,
